src/layers/CustomDisplayObjects/ComplexRopeGeometry.ts

```diff
@@ -0,0 +1,212 @@
+import { MeshGeometry } from 'pixi.js';
+import { sum, max } from 'd3-array';
+import { ComplexRopeSegment } from './ComplexRope';
+
+/**
+ * RopeGeometry allows you to draw a geometry across several points and then manipulate these points.
+ * @example
+ * import { RopeGeometry, Point } from 'pixi.js';
+ *
+ * for (let i = 0; i < 20; i++) {
+ *     points.push(new Point(i * 50, 0));
+ * };
+ * const rope = new RopeGeometry(100, points);
+ * @memberof PIXI
+ */
+export class ComplexRopeGeometry extends MeshGeometry {
+  /** An array of segments with points and diameter that determine the rope. */
+  public segments: ComplexRopeSegment[];
+
+  /** Rope texture scale. */
+  public readonly textureScale: number;
+
+  /**
+   * The width (i.e., thickness) of the rope.
+   * @readonly
+   */
+  _width: number;
+
+  /**
+   * @param width - The width (i.e., thickness) of the rope.
+   * @param points - An array of {@link PIXI.Point} objects to construct this rope.
+   * @param textureScale - scaling factor for repeated texture. To create a tiling rope
+   *     set baseTexture.wrapMode to {@link PIXI.WRAP_MODES.REPEAT} and use a power of two texture.
+   */
+  constructor(width = 200, segments: ComplexRopeSegment[], textureScale = 0) {
```

Comment:
Can we switch `width` and `segments` parameter order, as `segments` is the only one not having a default value.